### PR TITLE
feat(package/appwrite): update `appwrite` sdk to `15.0.0`

### PR DESCRIPTION
**Pull request by soranoo**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked #6090 
- [x] Tests for the changes have been added
- [x] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the new behavior?

fixes (issue)

### CODE REVIEW

File: packages/appwrite/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+### Major Changes
+
+#### Codebase
+- Replaced all GZIP string to JSON object.
+  - The JSON object will be transformed into GZIP string by `zlib` library during the test.
+  - This change is aimed to make the codebase (tests) more readable and maintainable.
+
+#### Dependency
+- Bump `appwrite` SDK from `10.0.1` to version `14.0.1`
+
```
Comment (BatuhanW) [inline packages/appwrite/CHANGELOG.md:14]: Please revert these changes. You **only** need to add changeset, we'll take care of the rest.

File: packages/appwrite/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@refinedev/appwrite",
-  "version": "6.5.3",
+  "version": "6.6.0",
```
Comment (BatuhanW) [inline packages/appwrite/package.json:3]: This change is also redundant.

File: .changeset/silver-crabs-relax.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+"@refinedev/appwrite": major
```
Comment (BatuhanW) [inline .changeset/silver-crabs-relax.md:2]:
```suggestion
"@refinedev/appwrite": patch
```

**Review by BatuhanW — CHANGES_REQUESTED** [review]
Hey @soranoo thanks for the PR, added some comments for changes.

**Comment by soranoo** [comment]
Done, plz take a look.

**Comment by soranoo** [comment]
Should I update the root deps?

**Comment by BatuhanW** [comment]
@soranoo you need to do pnpm install to update pnpm-lock.yaml. Also we'll provide new appwrite credentials, so we can record tests again. These tests doesn't ensure that it works correctly, just make assertions on already hardcoded mocks.

**Review by aliemir — COMMENTED** [review]
Hey @soranoo, is there any specific reason for you to stay in v14 instead of 15? 

There was some issues with our tests, I've updated them accordingly with fresh data. Also due to the changes in the SDK, updated one line in examples.

Also bumped Appwrite dependency to 15 🤔 Any reason behind the downgrade to 14? 

Thank you for your contribution! 🙏

**Comment by soranoo** [comment]
The package works with Appwrite SDK 15.0.0 but the tests don't. 

Here's the problem I am facing
https://github.com/refinedev/refine/issues/6090#issuecomment-2227038565

But you just got everything on the right track.
Thank you ❤️

**Review by alicanerdurmaz — APPROVED** [review]

**Pull request merged**